Traceback (most recent call last):
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/jupyter_cache/executors/utils.py", line 51, in single_nb_execution
    executenb(
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/asyncio/base_events.py", line 616, in run_until_complete
    return future.result()
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/tsauter/venv/totoro/lib/python3.8/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# Solve the heat equation
T_all, dz, integration = heat_equation_time(20, 40, 5)

# Create 2D mesh grid
# First create the y-axis values
y = np.arange(-20,0,dz)
# then the x-axis values
x = np.arange(integration)
# use the arrays to create a 2D-mesh
X, Y = np.meshgrid(x, y)

# Plot results on the mesh
plt.figure(figsize=(12,5))
plt.contourf(X,Y,T_all[::-1],25,origin='lower')

# Axis labels
plt.xlabel('Days')
plt.ylabel('Depth [m]')
plt.colorbar()
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mTypeError[0m                                 Traceback (most recent call last)
Input [0;32mIn [6][0m, in [0;36m<cell line: 2>[0;34m()[0m
[1;32m      1[0m [38;5;66;03m# Solve the heat equation[39;00m
[0;32m----> 2[0m T_all, dz, integration [38;5;241m=[39m [43mheat_equation_time[49m[43m([49m[38;5;241;43m20[39;49m[43m,[49m[43m [49m[38;5;241;43m40[39;49m[43m,[49m[43m [49m[38;5;241;43m5[39;49m[43m)[49m
[1;32m      4[0m [38;5;66;03m# Create 2D mesh grid[39;00m
[1;32m      5[0m [38;5;66;03m# First create the y-axis values[39;00m
[1;32m      6[0m y [38;5;241m=[39m np[38;5;241m.[39marange([38;5;241m-[39m[38;5;241m20[39m,[38;5;241m0[39m,dz)

Input [0;32mIn [5][0m, in [0;36mheat_equation_time[0;34m(depth, Nz, years)[0m
[1;32m     26[0m Tnew [38;5;241m=[39m T
[1;32m     28[0m [38;5;66;03m# 2D-Array containing the vertical profiles for all time steps (depth, time)[39;00m
[0;32m---> 29[0m T_all [38;5;241m=[39m [43mnp[49m[38;5;241;43m.[39;49m[43mzeros[49m[43m([49m[43m([49m[43mNz[49m[43m,[49m[38;5;28;43mint[39;49m[43m)[49m[43m)[49m
[1;32m     32[0m [38;5;66;03m# Time loop[39;00m
[1;32m     33[0m [38;5;28;01mfor[39;00m t [38;5;129;01min[39;00m [38;5;28mrange[39m(integration):

[0;31mTypeError[0m: 'type' object cannot be interpreted as an integer
TypeError: 'type' object cannot be interpreted as an integer

